src/main/java/com/jvm_bloggers/core/data_fetching/blog_posts/NewBlogPostStoringActor.java

```diff
@@ -20,28 +20,44 @@
 @RequiredArgsConstructor
 public class NewBlogPostStoringActor extends AbstractActor {
 
+    private static final String HTTPS_PREFIX = "https://";
+    private static final String HTTP_PREFIX = "http://";
+
     private final BlogPostRepository blogPostRepository;
     private final BlogPostFactory blogPostFactory;
 
     @Override
     public Receive createReceive() {
         return receiveBuilder().match(RssEntryWithAuthor.class,
             rssEntry -> {
-                if (Validators.isUrlValid(rssEntry.getRssEntry().getLink())) {
+                String blogPostLink = rssEntry.getRssEntry().getLink();
+                if (Validators.isUrlValid(blogPostLink)) {
                     BlogPost blogPost = blogPostRepository
-                            .findByUrl(rssEntry.getRssEntry().getLink())
+                            .findByUrl(blogPostLink)
+                            .orElse(() ->
+                                blogPostRepository
+                                    .findByUrl(createLinkWithSwitchedProtocol(blogPostLink))
```

Comment:
@tdziurko Instead of doing two queries to DB (in the worst case scenario) we could do just one call e.g. `blogPostRepository.findByUrlContaining("//www....")` [1].

[1] https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation